src/paperless/signals.py

```diff
@@ -30,3 +30,21 @@ def handle_failed_login(sender, credentials, request, **kwargs):
             log_output += f" from private IP `{client_ip}`."
 
     logger.info(log_output)
+
+
+def handle_social_account_updated(sender, request, sociallogin, **kwargs):
+    """
+    Handle the social account update signal.
+    """
+    from django.contrib.auth.models import Group
+
+    social_account_groups = sociallogin.account.extra_data.get(
```

Comment:
Assuming extra_data is a dict, `.get(key, [])` means social_account_groups will never be None.  I assume the filter just returns nothing and that ends up clearing the groups entirely

Reply:
Hmm, as I recall in my testing before I had it configured correctly it did return `None`. I'm not 100% certain but I thought it was more like "if has attribute use that (which can be None) if no attribute default to []", the difference being if you do want to sync "no groups"

(if that makes sense)